compiler/rustc_middle/src/ty/layout.rs

```diff
@@ -2590,9 +2590,20 @@ impl<'tcx> ty::Instance<'tcx> {
                 // `src/test/ui/polymorphization/normalized_sig_types.rs`), and codegen not keeping
                 // track of a polymorphization `ParamEnv` to allow normalizing later.
                 let mut sig = match *ty.kind() {
-                    ty::FnDef(def_id, substs) => tcx
-                        .normalize_erasing_regions(tcx.param_env(def_id), tcx.fn_sig(def_id))
-                        .subst(tcx, substs),
+                    ty::FnDef(def_id, substs) => {
+                        let fn_sig = tcx.fn_sig(def_id);
+                        match tcx.try_normalize_erasing_regions(tcx.param_env(def_id), fn_sig) {
+                            Ok(fn_sig) => fn_sig.subst(tcx, substs),
+                            Err(e) => {
+                                tcx.sess.delay_span_bug(
```

Comment:
still gotta delay a bug here, for the current test to pass.

Reply:
i still don't like that too much :sweat_smile:

Reply:
Hm, well if I remove this then we'll have an error message (the one I added) and then an ICE when `normalize_erasing_regions` fails. Would we rather have that? :thinking:

Reply:
no, my idea is to change the error during `normalize_erasing_regions` to be fatal, at which point we never get here if it failed because of `generic_const_exprs`